src/protocol/DataFeed.sol

```diff
@@ -38,6 +38,17 @@ contract DataFeed is IDataFeed {
         require(numBlobs > 0, "no data to publish");
 
         uint256 nQueries = queries.length;
+
+        uint256 totalValue;
+        bytes[] memory metadata = new bytes[](nQueries);
+        for (uint256 i; i < nQueries; ++i) {
+            metadata[i] = IPublicationHook(queries[i].provider).beforePublish{value: queries[i].value}(
+                msg.sender, queries[i].input
```

Comment:
shall we split the input into preInput and postInput?

Reply:
I think I may slight prefer an alternative implementation: adding a flag to the Query to indicate whether this query should run before or after the publication. So we have one not two hook functions in the hook interface:

```solidity

interface IPublicationHook {
function someName(address publisher, bool isPriorToPublication, bytes memory input) external payable returns (bytes memory output);
}

```